src/route/dataset-route.ts

```diff
@@ -39,72 +48,81 @@ function checkDatasetID(datasetID: string, res: Response): boolean {
-            ]
-        };
+            }
+        ]
+    };
+}
+
+
+apiRoute.post('/', upload.single('csv'), async (req: Request, res: Response) => {
+    if (!req.file) {
         res.status(400);
-        res.json(err);
+        res.json(errorDtoGenerator('csv', 'errors.no_csv_data'));
         return;
     }
     const lang: string = req.body?.language || req.i18n.language;
     const title: string = req.body?.title;
     if (!title) {
-        const err: ViewErrDTO = {
-            success: false,
-            dataset_id: undefined,
-            errors: [
-                {
-                    field: 'title',
-                    message: [
-                        {
-                            lang: ENGLISH,
-                            message: t('errors.no_title', { lng: ENGLISH })
-                        },
-                        {
-                            lang: WELSH,
-                            message: t('errors.no_title', { lng: WELSH })
-                        }
-                    ],
-                    tag: {
-                        name: 'errors.no_title',
-                        params: {}
-                    }
-                }
-            ]
-        };
         res.status(400);
-        res.json(err);
+        res.json(errorDtoGenerator('title', 'errors.no_title'));
         return;
     }
-    const dataset = Dataset.createDataset(title, 'BetaUser');
-    const saved_dataset_record = await dataset.save();
-    saved_dataset_record.addTitleByString(title, lang);
-    const uploadDTO = await uploadCSVToBlobStorage(req.file?.buffer, saved_dataset_record);
-    if (!uploadDTO.success) {
-        res.status(400);
+    let importRecord: Import;
+    try {
+        importRecord = await uploadCSVToBlobStorage(req.file?.stream, req.file?.mimetype);
+    } catch (err) {
+        logger.error(`An error occured trying to upload the file with the following error: ${e}`);
+        res.status(500);
+        res.json({ message: 'Error uploading file' });
+        return;
     }
+
+    // Everything looks good so far, let's create the dataset and revision records
+    const dataset = new Dataset();
+    dataset.creation_date = new Date();
+    // TODO change how we handle authentication to get the user on the Backend
+    dataset.created_by = 'Test User';
+    const saved_dataset_record = await dataset.save();
+    const datasetInfo = new DatasetInfo();
+    datasetInfo.language = lang;
+    datasetInfo.title = title;
+    datasetInfo.dataset = saved_dataset_record;
+    datasetInfo.save();
+    const revision = new DatasetRevision();
+    revision.dataset = saved_dataset_record;
+    revision.revision_index = 1;
+    revision.creation_date = new Date();
+    // TODO change how we handle authentication to get the user on the Backend
+    revision.created_by = 'Test User';
+    const saved_revision_record = await revision.save();
+    importRecord.revision = saved_revision_record;
+    importRecord.save();
+
```

Comment:
Commit all the objects here in a single database transaction.
Currently there doesn't seem to be a transaction at all, so we need to add one.
The code as written also makes several round-trips to the database server which is not advisable given that we're using a remote server rather than something in-process.
Therefore, batch up all the queries as well. There should be some kind of batching / pipelining mode in the database driver.